### PR TITLE
Add Stream Integration 4x Migration document

### PR DESCRIPTION
**Pull request by ruthryi**
E - 1692

### CODE REVIEW

File: en/docs/streaming/migrating-from-stream-processor.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+The following prerequisites should be completed before upgrading.
+
+- Make a backup of the SP 4.4.0 database and copy the <SP_HOME> directory in order to backup the product configurations.
+- Download the Streaming Integrator from the [Enterprise Integrator Home](https://wso2.com/integration/)
+
+!!! info "Updating the artifact IDs"
+    Note the following when migrating from WSO2 Streaming Processor(SP) to WSO2 Streaming Integrator(SI)
+    SP 4.x uses Siddhi 4.x and SI uses Siddhi 5.x from 1.x onwards. Therefore the group ID should be **renamed**
+    from `org.wso2.siddhi` to `io.siddhi`
```
Comment (senthuran16) [inline en/docs/streaming/migrating-from-stream-processor.md:17]:
```suggestion
!!! info "Migrating Custom Siddhi Extensions"
    Note the following when migrating custom Siddhi extensions from WSO2 Stream Processor(SP) to WSO2 Streaming Integrator(SI).
    SP 4.x uses Siddhi 4.x and SI uses Siddhi 5.x from 1.x onwards. Therefore the group ID and package names should be **renamed** from `org.wso2.siddhi` to `io.siddhi`
```

**Review by senthuran16 — DISMISSED** [review]

File: en/docs/streaming/migrating-from-stream-processor.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Migrating from WSO2 Stream Processor
+
+The Streaming Integrator performs all functions that are also performed by [WSO2 Stream Processor](https://docs.wso2.com/display/SP440/Stream+Processor+Documentation). It also has additional features to trigger integration flows in order to take action in response to results derived after analyzing data.
+
+If you are currently using WSO2 Stream Processor to carry out any streaming integration/stream processing activities and want to carry them out in the Streaming Integrator, you can migrate your setup as follows:
+
+## Preparing to upgrade
```
Comment (Mariangela) [inline en/docs/streaming/migrating-from-stream-processor.md:7]: Is it better to put the sub-heading as Step 1,2,3 etc. Example, 
- Before you begin
- Step 1 - Migrate the databases

File: en/docs/streaming/migrating-from-stream-processor.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Note the following when migrating from WSO2 Streaming Processor(SP) to WSO2 Streaming Integrator(SI)
+    SP 4.x uses Siddhi 4.x and SI uses Siddhi 5.x from 1.x onwards. Therefore the group ID should be **renamed**
+    from `org.wso2.siddhi` to `io.siddhi`
+
+## Migrating Databases
+
+To connect the Streaming Integrator to the same databases as WSO2 SP 4.4.0 so that the persisted data can be accessed, configure the data sources as follows:
+
+- Configure the data sources in the `<SI_HOME>/conf/server/deployment.yaml` file the same way you have configured them in `<SP_HOME>/conf/wso2/worker/deployment.yaml` file.
+- Configure the data sources in the `<SI__TOOLING_HOME>/conf/server/deployment.yaml` file the same way you have configured them in `<SP_HOME>/conf/wso2/editor/deployment.yaml` file.
```
Comment (Mariangela) [inline en/docs/streaming/migrating-from-stream-processor.md:24]: There is an extra underscore in `<SI__TOOLING_HOME>`

File: en/docs/streaming/migrating-from-stream-processor.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    !!!info
+        The Business Rules feature which was a part of the `Dashboard` profile of the Stream Processor is now shipped with Streaming Integrator Tooling. Therefore, configurations related to this feature are added in the `<SI__TOOLING_HOME>/conf/server/deployment.yaml` file.
+
+For the complete list of data sources configured for the Streaming Integrator, see [Configuring Data sources](https://ei.docs.wso2.com/en/latest/streaming-integrator/setup/configuring-data-sources/).
+
+## Migrating Siddhi applications
+
+To migrate the Siddhi applications that you have deployed in WSO2 SP 4.4.0, follow the procedure below:
+
+1. Copy all the Siddhi applications in the `<SP_HOME/wso2/worker/deployment/siddhi-files` directory.
```
Comment (Mariangela) [inline en/docs/streaming/migrating-from-stream-processor.md:36]: <SP_HOME should be <SP_HOME>

File: en/docs/streaming/migrating-from-stream-processor.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+2. Place the Siddhi applications you copied in the `<SI_HOME/wso2/server/deployment/siddhi-files` directory.
```
Comment (Mariangela) [inline en/docs/streaming/migrating-from-stream-processor.md:38]: <SI_HOME should be <SI_HOME>

File: en/docs/streaming/migrating-from-stream-processor.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+!!! note
+    Note that some extensions are not packed by default when migrating from SP to SI. You can find these extensions in the `SP_HOME/lib` directory, and copy them as required.
```
Comment (Mariangela) [inline en/docs/streaming/migrating-from-stream-processor.md:41]: This SP_HOME/lib should be <SP_HOME>/lib

File: en/docs/streaming/migrating-from-stream-processor.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    SP 4.x uses Siddhi 4.x and SI uses Siddhi 5.x from 1.x onwards. Therefore the group ID should be **renamed**
+    from `org.wso2.siddhi` to `io.siddhi`
+
+## Migrating Databases
+
+To connect the Streaming Integrator to the same databases as WSO2 SP 4.4.0 so that the persisted data can be accessed, configure the data sources as follows:
+
+- Configure the data sources in the `<SI_HOME>/conf/server/deployment.yaml` file the same way you have configured them in `<SP_HOME>/conf/wso2/worker/deployment.yaml` file.
+- Configure the data sources in the `<SI__TOOLING_HOME>/conf/server/deployment.yaml` file the same way you have configured them in `<SP_HOME>/conf/wso2/editor/deployment.yaml` file.
+- Check the data source configured for Business Rules  in the `<SP_HOME>/conf/wso2/dashboard/deployment.yaml` file, and configure that data source with the same parameter values in the `<SI__TOOLING_HOME>/conf/server/deployment.yaml` file.
```
Comment (Mariangela) [inline en/docs/streaming/migrating-from-stream-processor.md:25]: double spaces after "Business Rules"  

"Check the data source configured for Business Rules  in the ........"

**Review by Mariangela — COMMENTED** [review]
@ruthryi , please apply the changes based on the comments for the other related PRs as well.

**Review by Mariangela — APPROVED** [review]

**Comment by Mariangela** [comment]
Checked the PR locally and reviewed and updated it. The following is the output after the changes.
screen-shot

**Review by Mariangela — APPROVED** [review]

**Pull request merged**